model_managers/model_manager.py

```diff
@@ -291,13 +291,13 @@ def create_relationship(self, rel_list: List[List[str]], identifier='label', mat
         SET rel_node.`FROM.Class.label` = from.label
         SET rel_node.`TO.Class.label` = to.label
         RETURN collect([from.`{identifier}`, to.`{identifier}`, rel_node.relationship_type]) as rels", 
-        "WITH rel[0] as from_identity, rel[1] as to_identity, rel[2] as rel_type, rel[3] as optional
+        "WITH rel[0] as from_identity, rel[1] as to_identity, rel[2] as rel_type, rel[3] as optional, rel[4] as propagated_from
         {'MATCH' if match_classes else 'MERGE'} (from:Class {{`{identifier}`:from_identity}})
         {'MATCH' if match_classes else 'MERGE'} (to:Class {{`{identifier}`:to_identity}})   
-        MERGE (from)<-[:FROM]-(rel_node:Relationship{{relationship_type:rel_type, relationship_optional:optional}})-[:TO]->(to)
+        MERGE (from)<-[:FROM]-(rel_node:Relationship{{relationship_type:rel_type, relationship_optional:optional, relationship_propagated_from:propagated_from}})-[:TO]->(to)
         SET rel_node.`FROM.Class.label` = from.label
         SET rel_node.`TO.Class.label` = to.label
-        RETURN collect([from.`{identifier}`, to.`{identifier}`, rel_node.relationship_type, rel_node.relationship_optional]) as rels
+        RETURN collect([from.`{identifier}`, to.`{identifier}`, rel_node.relationship_type, rel_node.relationship_optional, rel_node.relationship_propagated_from]) as rels
         ",
```

Comment:
@Vidhisri56 I think the 'relationship_propagated_from' property should be added irrespective of whether the optional property is there or not. It should be there in both the conditions when size<3 or >3. Let me know if there is any confusion.